enclave/src/polkadex_cache/cancel_order_cache.rs

```diff
@@ -96,3 +96,115 @@ impl CancelOrderCache {
         self.request_id = self.request_id.saturating_add(1)
     }
 }
+
+
+
+pub mod tests {
+    use super::*;
+    use codec::Encode;
+
+
+    pub fn test_initialize_and_lock_storage() {
+        // given
+        CancelOrderCache::initialize();
+
+        // when
+        let mutex = CancelOrderCache::load().unwrap();
+
+        // then
+        mutex.lock().unwrap();
+    }
+
+    pub fn test_insert_order_and_increment() {
+        // given
+        CancelOrderCache::initialize();
+        let mut cache = CancelOrderCache::load()
+            .unwrap()
+            .lock()
+            .unwrap();
+        let order_uuid: OrderUUID = "hello_world".encode();
+        assert_eq!(cache.request_id(), 0);
+
+        // when
+        assert!(cache.insert_order(order_uuid.clone()));
+
+        // then
+        assert_eq!(cache.request_id(), 1);
+        assert!(cache.contains(&order_uuid));
+    }
+
+    /// inserts two orders
+    /// removes the second, but leaves the first
+    /// then checks if second was really removed
+    pub fn test_remove_order() {
```

Comment:
this is nice to read! It's either commenting the test like you did, or adding the description in the test name, e.g.
```
given_two_orders_when_removing_one_then_first_should_remain()
```
but it can result in veery long function names, which is not always helpful